mongodb/src/main/java/org/apache/metamodel/mongodb/MongoDbDataContext.java

```diff
@@ -435,6 +441,19 @@ private String getOperatorName(FilterItem item) {
         return operatorName;
     }
 
+    private Pattern turnOperandIntoRegExp(Object operand) {
+        String operandAsString = operand.toString();
+        StringBuilder operandAsRegExp = new StringBuilder(replaceWildCardLikeChars(operandAsString));
+        if (!operandAsString.startsWith("%") && operandAsString.contains("%")) {
+            operandAsRegExp = operandAsRegExp.insert(0, "^");
+        }
+        return Pattern.compile(operandAsRegExp.toString(), Pattern.CASE_INSENSITIVE);
+    }
+
+    private String replaceWildCardLikeChars(String operand) {
+        return operand.replace("%","");
```

Comment:
Yes, you are right, is just removing the %. I think we don't need to add more complexity to the regular expression, when we'd receive a string like '%MALE%' we just need to create a pattern like this (removing the %):
```
Pattern j = Pattern.compile("MALE", Pattern.CASE_INSENSITIVE);
```
Whereas when we'd receive a string like 'MALE%' the pattern should be (removing % and prepend ^):
```
Pattern j = Pattern.compile("^MALE", Pattern.CASE_INSENSITIVE);
```
I think these are the only cases we should care about. Both are working fine as tested in the test I've added.

What do you think?